includes/class-webmention-avatar-handler.php

```diff
@@ -12,140 +12,224 @@ public static function init() {
 		$cls = get_called_class();
 
 		add_filter( 'pre_get_avatar_data', array( $cls, 'avatar_stored_in_comment' ), 30, 2 );
-		add_filter( 'get_avatar_data', array( $cls, 'anonymous_avatar_data' ), 30, 2 );
 
-		// All the default gravatars come from Gravatar instead of being generated locally so add a local default
-		add_filter( 'avatar_defaults', array( $cls, 'anonymous_avatar' ) );
+		// Store Avatars Locally
+		add_action( 'comment_post', array( $cls, 'store_avatar' ), 20 );
+		add_action( 'edit_comment', array( $cls, 'store_avatar' ), 20 );
 	}
 
-	public static function anonymous_avatar( $avatar_defaults ) {
-		$avatar_defaults['mystery'] = __( 'Mystery Person (hosted locally)', 'webmention' );
-
-		return $avatar_defaults;
+	/**
+	 * Return upload directory.
+	 *
+	 * @param string $filepath File Path. Optional
+	 * @return string URL of upload directory.
+	 */
+	public static function upload_directory( $filepath = '' ) {
+		$upload_dir = wp_get_upload_dir();
+		$upload_dir = $upload_dir['basedir'] . '/webmention/avatars/';
+		$upload_dir = apply_filters( 'webmention_avatar_directory', $upload_dir );
+		return $upload_dir . $filepath;
 	}
 
 	/**
-	 * Function to retrieve Avatar if stored in meta
+	 * Return upload directory url.
 	 *
+	 * @param string $filepath File Path. Optional.
+	 * @return string URL of upload directory.
+	 */
+	public static function upload_directory_url( $filepath = '' ) {
```

Comment:
isn't that exactly the same code as in the `upload_directory` function?

Reply:
One should return the URL, the other the filepath. So, for example.

/var/html/wordpress/wp-content/uploads/webmention/avatar/

vs

https://example.com/uploads/webmention/avatar

Reply:
Ah I see, one uses `baseurl` and the other `basedir`. But both are using the same filter?

Reply:
because it shares a lot of the same code, shouldn't we add a helper function with two params... the type and the filepath?

Reply:
Good point, I'll adjust.

Reply:
Do you have any thoughts on the md5 vs sha256 issue?

Reply:
Lets use md5... this is nothing security relevant, so no problem to use it, in my opinion.

Reply:
I'll redo the function. I tested and it does work perfect and migrate any time the comment is updated.